File: backend/app/schemas/fraud_analysis.py
"""fraud_analysis.py
레이어: Schemas
역할: 사기 화면 탐지(fraud_analysis) 기능의 API Response 데이터 계약을 정의한다.
"""

from typing import Literal

from pydantic import BaseModel


class FraudAnalysisResult(BaseModel):
    """Claude Vision 분석 결과를 표현하는 Response Schema이다."""

    verdict: Literal["정상", "사기의심", "판단불가"]
    tamper_types: list[str]
    reasoning: str
    confidence: float
    undetermined_reason: str | None = None

File: frontend/types/fraud_analysis.ts
// 목적: 사기화면 판별(fraud-analysis) API 응답 타입을 정의한다.
// 주요 역할: Backend Pydantic Schema(FraudAnalysisResult)와 동일한 필드명·의미를 유지한다.
//           API JSON 필드는 snake_case 규칙을 유지한다.

export type FraudVerdict = "정상" | "사기의심" | "판단불가";

export interface FraudAnalysisResult {
  verdict: FraudVerdict;
  tamper_types: string[];
  reasoning: string;
  confidence: number;
  undetermined_reason: string | null;
}
